test/testdata/rbi/kernel.rb

```diff
@@ -65,9 +65,16 @@ def raise_test
 obj = T.let("foo", String)
 T.assert_type!(obj.itself, String)
 
-y = loop do
-end
-puts y # error: This code is unreachable
+
+# These types are deliberately wrong, because `Kernel#p` is difficult to type
+# in an RBI.  See the comments in kernel.rbi.
+p_result = Kernel.p 1
+T.reveal_type(p_result) # error: Revealed type: `NilClass`
+p_result = p "string"
+T.reveal_type(p_result) # error: Revealed type: `NilClass`
+# This should actually be typed as an array.
```

Comment:
Nit: Could you add a newline between each case? At first I thought this comment was referring to line 74 and was confused.